osquery/tables/system/windows/objects.cpp

```diff
@@ -0,0 +1,229 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <Windows.h>
+
+#include <osquery/core.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/windows/ntapi.h"
+#include "osquery/core/windows/wmi.h"
+#include "osquery/core/conversions.h"
+#include "osquery/core/windows/handle.h"
+
+namespace osquery {
+namespace tables {
+
+// the windows object namespace consists of nested directories,
+// beginning with \
+//
+// each directory consists of zero or more entries, each of which consists
+// of an unqualified object name and an object type name.  both are
+// represented by the windows kernel as UTF-16 strings
+//
+// the pair of object name and object type name are used frequently
+// in this module.  use a std::pair of std::wstrings to represent this
+// pair and provide a simpler name
+//
+// this type definition is internal to this module and meant as a
+// type to be used with the rest of osquery.  this note is important
+// because osquery uses UTF-8 encoded std::strings throughout.  Because
+// windows uses UTF-16 strings, which are most easily represented by
+// std::wstring, std::wstrings are used internally to this module and
+// converted to UTF-8 encoded std::strings when communicating with the
+// rest of osquery
+//
+typedef std::pair<std::wstring, std::wstring> obj_name_type_pair;
+
+// arbitrary upper bound on number of supported objects in a single
+// directory to query.  windows provides a means to query a single object
+// at a time via an index; this arbitrary upper bound is a backstop
+//
+static const unsigned long max_supported_objects = 1024 * 1024;
+
+// arbitrary buffer size to be used when querying for a single
+// object-name--object-type-name pair with NtQueryDirectoryObject
+//
+static const unsigned long obj_buf_size = 8 * 1024;
```

Comment:
`osquery` has a bit different way to name global consts:
```c++
static const unsigned long kObjBufSize = 8 * 1024;
```

Reply:
@akindyakov @muffins got it, will address.

Reply:
done